torch/futures/__init__.py

```diff
@@ -138,3 +143,21 @@ def wait_all(futures):
         :class:`~torch.futures.Future` throws.
     """
     return [fut.wait() for fut in torch._C._collect_all(futures).wait()]
+
+
+try:
+
+    class Future(PyFuture, GenericWithOneTypeVar):
+        # Combine the implementation class and the type class.
+        pass
+
+
+except TypeError as exc:
+    # TypeError: metaclass conflict: the metaclass of a derived class
+    # must be a (non-strict) subclass of the metaclasses of all its bases
+    class FutureMeta(PyFuture.__class__, GenericWithOneTypeVar.__class__):
+        pass
+
+    class Future(PyFuture, GenericWithOneTypeVar, metaclass=FutureMeta):
```

Comment:
Curious, why not always use this approach instead of first trying `class Future(PyFuture, GenericWithOneTypeVar)`? Is there any downside of the fallback path?

Reply:
These 2 paths are actually exclusive to each other. I haven't figured out what is the good test condition to decide which path to use.